technologydata/sources.py

```diff
@@ -365,85 +354,96 @@ def download_file_from_wayback(self) -> list[pathlib.Path | None]:
         Notes
         -----
         - The `details` attribute must contain a key "url_archived" with a valid URL.
-        - The `path` and `name` attributes must be defined in the instance for saving the file.
+
+        Examples
+        --------
+        >>> from technologydata import Source
+        >>> source = Source("example01", "./some/local/path/")
+        >>> storage_paths = source.download_file_from_wayback(pathlib.Path("base_path"))
 
         """
-        if self.details is None:
+        if self.details is None or self.details.empty:
             logger.error(f"The details attribute of the source {self.name} is not set.")
-            return []
+            return {}
         if (
             "url_archived" not in self.details
             or self.details["url_archived"].isna().all()
         ):
             logger.error(
                 f"The url_archived attribute of source {self.name} is not set."
             )
-            return []
-        if self.path is None:
-            logger.error(f"The path attribute of the source {self.name} is not set.")
-            return []
+            return {}
+        if download_directory is None:
+            logger.error(f"The base path of the source {self.name} is not set.")
+            return {}
 
-        saved_paths: list[pathlib.Path | None] = []  # Explicit type annotation
+        saved_paths: dict[str, pathlib.Path | None] = {}  # Explicit type annotation
         for index, row in self.details.iterrows():
             url_archived = row["url_archived"]
-            save_path = self._get_save_path(url_archived)
+            source_title = td.Utils.replace_special_characters(row["title"])
+            save_path = self._get_save_path(
+                url_archived, download_directory, source_title
+            )
 
             if save_path is None:
-                saved_paths.append(None)
+                logger.debug(
+                    f"It was not possible to determine a file path for the source {source_title}."
+                )
                 continue
+
+            if save_path in saved_paths.values() or save_path.is_file():
+                logger.debug(f"There is already a file stored at the path {save_path}.")
```

Comment:
```suggestion
                logger.warning(f"There is already a file stored at the path {save_path}. Not downloading or overwriting this file.")
```

similarly - we're not downloading something again or overwriting anything